File: chat-tiempo-real-offline/frontend/chat-tiempo-real-offline/src/app/models/auth.model.ts
/*
  Modelos de autenticación.

  LoginRequest:
  Datos que el usuario escribe en el formulario.

  LoginResponse:
  Datos que devuelve el backend si el inicio de sesión es correcto.
*/

import { Usuario } from './usuario.model';

export interface LoginRequest {
  username: string;
  password: string;
}

export interface LoginResponse {
  token: string;
  usuario: Usuario;
}

File: chat-tiempo-real-offline/backend/models/auth_model.py
"""
auth_model.py

Modelos relacionados con autenticación.

LoginRequest:
Datos que envía Angular al iniciar sesión.

LoginResponse:
Datos que devuelve el backend cuando el login es correcto.
"""

from pydantic import BaseModel
from models.usuario_model import Usuario


class LoginRequest(BaseModel):
    username: str
    password: str


class UsuarioResponse(BaseModel):
    id: int
    nombre: str
    username: str


class LoginResponse(BaseModel):
    token: str
    usuario: UsuarioResponse